### PR TITLE
Clarify interval log semantics.

### PR DESCRIPTION
**Pull request by marshallpierce**
See https://gitter.im/HdrHistogram/HdrHistogram?at=5a1a731771ad3f87362ebd42.

Who knows if anyone will ever read all of this, but the lack of clarity was bothering me. ;)

### CODE REVIEW

File: src/serialization/interval_log/mod.rs [hunk 1]
```diff
@@ -12,58 +12,102 @@
...
+//! some cases log files are consumed by humans too.
+//!
+//! While BaseTime is used to redefine per-interval timestamps for human readability, StartTime
+//! provides information about when the process that is generating the log reached some "start"
+//! condition. It's frequently the case that intervals will start some seconds after the process
+//! started, whether due to initial warmup before a benchmark or because it just takes a
+//! while to start up. If this is relevant to your workload, use StartTime to record the actual
+//! process start time (or other relevant "start" condition, like when a benchmark begins on an
+//! already long-running process). You could then use this when processing a log to more accurately
+//! plot interval data over time.
```
Comment (jonhoo) [inline src/serialization/interval_log/mod.rs]: Are the per-interval timestamps then relative to `StartTime`?

Comment (jonhoo) [reply src/serialization/interval_log/mod.rs]: From reading the example below, I take it they aren't (that is, they are *always* relative to `BaseTime` or UNIX epoch). We might want to clarify that here.

File: src/serialization/interval_log/mod.rs [hunk 2]
```diff
@@ -12,58 +12,102 @@
...
+//!
+//! - Neither StartTime nor BaseTime are present: interval timestamps are interpreted as seconds
+//! since the epoch. The first interval's timestamp is stored to the StartTime field.
 //! - StartTime is present: StartTime is a number of seconds since epoch, and interval timestamps
-//! should be interpreted as deltas that could be added to StartTime if seconds since epoch for each
-//! interval is needed.
-//! - BaseTime is present: same as the case where StartTime is present. It's seconds since epoch,
-//! with interval timestamps as deltas.
+//! may be interpreted as deltas to be added to StartTime or as "absolute" Unix time depending on a
+//! heuristic. In other words, the heuristic chooses between setting the effective BaseTime to 0 or
```
Comment (jonhoo) [inline src/serialization/interval_log/mod.rs]: Is this heuristic straightforward to explain?

File: src/serialization/interval_log/mod.rs [hunk 3]
```diff
@@ -12,58 +12,102 @@
...
+//! to StartTime.
+//! - BaseTime is present: BaseTime is a number of seconds since epoch, and interval timestamps are
+//! interpreted as deltas. The first interval's timestamp is stored to the StartTime field.
```
Comment (jonhoo) [inline src/serialization/interval_log/mod.rs]: Hmm.. So `StartTime` is also relative to `BaseTime`? Sort of?

Comment (marshallpierce) [reply src/serialization/interval_log/mod.rs]: This kind of thing is why I didn't want to make a higher level iterator that's an equivalent to HistogramLogReader until we get some clarity from users about what parts of its behavior they actually use. :)

**Comment by jonhoo** [comment]
LGTM!

**Pull request merged**